core/src/main/java/org/elasticsearch/action/admin/indices/template/put/PutIndexTemplateRequestBuilder.java

```diff
@@ -41,7 +42,7 @@ public PutIndexTemplateRequestBuilder(ElasticsearchClient client, PutIndexTempla
     /**
      * Sets the template match expression that will be used to match on indices created.
      */
-    public PutIndexTemplateRequestBuilder setTemplate(String template) {
+    public PutIndexTemplateRequestBuilder setTemplate(List<String> template) {
```

Comment:
You could add a new method that takes a String does a `singletoneList` on the input for backwards compatibility but we don't tend to do that for these builders because we expect that users of the Java API can deal with these changes on upgrade because the compiler will catch it for them. In other words: we don't respect semver in the Java API.